packages/shared/world/parcelSceneManager.ts

```diff
@@ -122,99 +145,260 @@ function reportPendingScenes() {
+    unloadParcelSceneById(parcelSceneLoadingState.isolatedModeOptions?.payload.sceneId)
+    parcelSceneLoadingState.desiredParcelScenes.delete(parcelSceneLoadingState.isolatedModeOptions?.payload.sceneId)
+  }
+  // We deactivate the isolated mode
+  parcelSceneLoadingState.runningIsolatedMode = false
+  parcelSceneLoadingState.isolatedModeOptions = null
+
+  //In the builder we need to go back to the world
+  if (options.mode === IsolatedMode.BUILDER) {
+    // We do a teleport to go back to the world to the last position of the player
+    let text = 'Going back to the world'
+    teleportObservable.notifyObservers({
+      x: lastPlayerPositionKnow.x,
+      y: lastPlayerPositionKnow.y,
+      text: text
+    })
+  }
+}
+
+Object.assign(globalThis, {
+  startIsolatedMode,
+  endIsolatedMode: stopIsolatedMode
+})
+
+/**
+ * Returns a set of Set<SceneId>
+ */
+function getDesiredParcelScenes(): Set<string> {
+  return new Set(parcelSceneLoadingState.desiredParcelScenes)
+}
+
+/**
+ * Receives a set of Set<SceneId>
+ */
+function setDesiredParcelScenes(desiredParcelScenes: Set<string>) {
+  const previousSet = new Set(parcelSceneLoadingState.desiredParcelScenes)
+  const newSet = (parcelSceneLoadingState.desiredParcelScenes = desiredParcelScenes)
+
+  // react to changes
+  for (const oldSceneId of previousSet) {
+    if (!newSet.has(oldSceneId) && loadedSceneWorkers.has(oldSceneId)) {
+      // destroy old scene
+      unloadParcelSceneById(oldSceneId)
+    }
+  }
+
+  for (const newSceneId of newSet) {
+    if (!loadedSceneWorkers.has(newSceneId)) {
+      // create new scene
+      loadParcelSceneByIdIfMissing(newSceneId).catch(sceneManagerLogger.error)
+    }
+  }
+}
+
+function unloadParcelSceneById(sceneId: string) {
+  const worker = loadedSceneWorkers.get(sceneId)
+  if (!worker || !parcelSceneLoadingState.lifecycleManager) {
+    return
+  }
+  //We notify that the scene has been unloaded, the sceneId must have the same name
+  parcelSceneLoadingState.lifecycleManager.notify('Scene.status', {
+    sceneId: sceneId,
+    status: 'unloaded'
   })
+  stopParcelSceneWorker(worker)
+  onUnloadParcelScenesObservable.notifyObservers([sceneId])
+}
+
+async function loadParcelSceneByIdIfMissing(sceneId: string) {
+  const lifecycleManager = parcelSceneLoadingState.lifecycleManager
+
+  if (!lifecycleManager) return
+
+  const parcelSceneToStart = await lifecycleManager.getParcelData(sceneId)
 
-  ret.on('Scene.shouldStart', async (opts: { sceneId: string }) => {
-    const sceneId = opts.sceneId
-    const parcelSceneToStart = await ret.getParcelData(sceneId)
+  // create the worker if don't exis
+  if (!getSceneWorkerBySceneID(sceneId)) {
+    //If we are running in isolated mode and it is builder mode, we create a stateless worker instead of a normal worker
+    if (
+      parcelSceneLoadingState.runningIsolatedMode &&
+      parcelSceneLoadingState.isolatedModeOptions?.mode === IsolatedMode.BUILDER
+    ) {
+      let sceneId: string = ''
```

Comment:
same here, this could be a const with a ternary so we avoid reading an if with a mutation after